text/0000-address.md

```diff
@@ -0,0 +1,231 @@
+   current RFC only proposes 2 models: simple and grouped.
+4. we also propose a new human readable encoding. This will allow visual
+   discrimination for the user between the old address format and the new one.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+An `Address` is a mean to associate a token to a cryptographic key pair.
+In order for Bod to send a token to Alice, Alice needs to generate a key
+pair, a `PublicKey` and a `PrivateKey`. Alice generates an `Address` from
+this `PublicKey` and sends the `Address` to Bob and Bob adds on the Ledger
+that he has sent a token to the public key.
+
+To create a new address, we only need the following information:
+
+* `PublicKey`: this will be necessary to be able to utilise the token;
+* `Discriminant`: this will prevent misuse of addresses between testing
+  environment and production environment;
+
+This is the most simple address. It is called a **Single Address**. This is
+because it is not associated with any staking key.
+
+It is possible to create a **Grouped Address**, it groups addresses to a
+staking key. To create a **Grouped Address** you need the `PublicKey`,
+the `Discriminant` (just like for a **Single Address**) and another
+`PublicKey` that is associating this address to another hierarchy.
+
+The human readable encoding of the `Address` is [bech32]. It includes
+a checksum for correctness (preventing mistyping the address). There
+is also a visual discriminant for human readable discrimination between
+cardano (production) addresses and testing version of the blockchain:
+
+* `ta` is the visual discriminant for **Testing** address;
+* `ca` is the visual discriminant for **Cardano** address (production address);
+
+So to create a **Single Address** we need a public key.
+
+```
+PublicKey = 0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20
+```
+
+A production **Single Address** would look like:
+`ca1qvqsyqcyq5rqwzqfpg9scrgwpugpzysnzs23v9ccrydpk8qarc0jqxuzx4s`. Note the `ca` at
+the beginning of the `Address`. This is our visual discriminant.
+
+To create a **Grouped Address** we need an additional grouping key:
+
+```rust
+GroupKey = 292a2b2c2d2e2f303132333435363738393a3b3c3d3e3f404142434445464748
+```
+
+A production **Grouped Address** would look like:
+`ca1qsqsyqcyq5rqwzqfpg9scrgwpugpzysnzs23v9ccrydpk8qarc0jq2f29vkz6t30xqcnyve5x5mrwwpe8ganc0f78aqyzsjrg3z5v36gguhxny`.
+Note the `ca` prefix again.
+
+A test **Grouped Address** would look like:
+`ta1ss5j52ev95hz7vp3xgengdfkxuurjw3m8s7nu06qg9pyx3z9ger5sqgzqvzq2ps8pqys5zcvp58q7yq3zgf3g9gkzuvpjxsmrsw3u8eqx5x7xh`.
+Note the `ta` prefix here.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Binary serialization
+
+It uses a simple serialization format which is made to be concise:
+
+* First byte contains the discrimination information (1 bit) and the kind of address (7 bits)
```

Comment:
Is the discriminant the most or least significant bit?

Reply:
It is specified again in the example below but can be specified here again to avoid confusion.

|discriminant | first byte | public key |
|---:|:---|:--|
| Production | `0x03` (`0b0000_0011`) | `0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20` |
| Testing    | `0x83` (`0b1000_0011`) | `0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20` |